doc/source/tutorial-quickstart-huggingface.rst

```diff
@@ -17,65 +17,90 @@ The end goal is to detect if a movie rating is positive or negative.
 Dependencies
 ------------
 
+First of all, it is recommended to create a virtual environment and run everything within a :doc:`virtualenv <contributor-how-to-set-up-a-virtual-env>`.
+
 To follow along this tutorial you will need to install the following packages:
-:code:`datasets`, :code:`evaluate`, :code:`flwr`, :code:`torch`, and :code:`transformers`.
+:code:`evaluate`, :code:`flwr`, :code:`flwr-datasets`, :code:`torch`, and :code:`transformers`.
 This can be done using :code:`pip`:
 
 .. code-block:: shell
 
-    $ pip install datasets evaluate flwr torch transformers
+    $ pip install evaluate flwr flwr-datasets torch transformers
+
+
+Flower Client
+-------------
+Now that we have all our dependencies installed, let's run a simple distributed training with two clients and one server.
+In a file called :code:`client.py`, import Flower and PyTorch related packages:
 
+.. code-block:: python
+
+    from collections import OrderedDict
 
-Standard Hugging Face workflow
-------------------------------
+    import flwr as fl
+    import torch
+    from evaluate import load as load_metric
+    from flwr.client import Client, ClientApp
+    from flwr_datasets import FederatedDataset
+    from torch.optim import AdamW
+    from torch.utils.data import DataLoader
+    from transformers import (
+        AutoModelForSequenceClassification,
+        AutoTokenizer,
+        DataCollatorWithPadding,
+    )
+
+In addition, we define the device allocation in PyTorch with:
+
+.. code-block:: python
+
+    DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+We also specify the model checkpoint:
+
+.. code-block:: python
+
+    CHECKPOINT = "distilbert-base-uncased"
 
 Handling the data
 ^^^^^^^^^^^^^^^^^
 
-To fetch the IMDB dataset, we will use Hugging Face's :code:`datasets` library.
+To fetch the IMDB dataset, we will use `Flower Datasets <https://flower.ai/docs/datasets/>`_.
+The :code:`FederatedDataset()` module downloads and partitions the dataset.
 We then need to tokenize the data and create :code:`PyTorch` dataloaders,
 this is all done in the :code:`load_data` function:
 
 .. code-block:: python
 
-    import random
-    import torch
-    from datasets import load_dataset
-    from torch.utils.data import DataLoader
-    from transformers import AutoTokenizer, DataCollatorWithPadding
+    def load_data(partition_id):
+        """Load IMDB data (training and eval)"""
+        fds = FederatedDataset(dataset="imdb", partitioners={"train": 1_000})
+        partition = fds.load_partition(partition_id)
```

Comment:
@adam-narozniak correct me if I'm wrong, but there is no more `partition_id` argument in `load_partition` right?

Reply:
@danielnugraha Linking my comment from the other PR [here](https://github.com/adap/flower/pull/3340#discussion_r1622280045). Also, in this code the variable name doesn't affect it since we're passing `partition_id` as a positional argument (here there's only 1 argument).

Reply:
Closing stale discussion thread.